ui/src/app/components/toast-banner.tsx

```diff
@@ -61,40 +60,42 @@ export enum ToastType {
   WARNING
 }
 
-const styleForType = (tt: ToastType) => switchCase(tt,
-  [ToastType.INFO, () => styles.infoBanner],
+const styleForType = (toastType: ToastType, zIndex): React.CSSProperties => switchCase(toastType,
+  [ToastType.INFO, () => ({
+    ...styles.infoBanner,
+    zIndex
+  })],
   [ToastType.WARNING, () => ({
-      ...styles.infoBanner,
-      backgroundColor: colorWithWhiteness(colors.highlight, .5),
+    ...styles.infoBanner,
+    zIndex,
+    backgroundColor: colorWithWhiteness(colors.highlight, .5),
   })],
 );
 
 interface ToastProps {
   title: string;
-  message: string;
+  message: string | JSX.Element;
   onClose: Function;
-  type: ToastType;
-  footer?: JSX.Element;
-}
-
-const InnerComponent = (props: ToastProps) => {
-  const {title, message, onClose, type, footer} = props;
-  return <FlexColumn style={styleForType(type)}>
-    <FlexRow style={{alignItems: 'center', marginTop: '.1rem'}}>
-      {type === ToastType.WARNING && warningIcon}
-      <div style={styles.title}>{title}</div>
-    </FlexRow>
-    <div style={styles.message}>{message}</div>
-    <div style={styles.footer}>{footer}</div>
-    <ClrIcon
-      shape={'times'}
-      size={20}
-      style={styles.closeIcon}
-      onClick={() => onClose()}
-    />
-  </FlexColumn>;
+  toastType: ToastType;
+  zIndex: any;  // TODO better type
```

Comment:
zIndex is now required, to help define an ordering if there are multiple banners